### PR TITLE
[CIAS30-3636] Delete predefined users when clearing intervention data

### PR DESCRIPTION
**Pull request by OskarDybowskiHTD**
## Related tasks
- [CIAS-3636](https://htdevelopers.atlassian.net/browse/CIAS30-3636)

## What's new?
- Predefined participants will now be cleared alongside their `PredefinedUserParameter` when clearing data for an intervention

### CODE REVIEW

File: spec/jobs/data_clear_jobs/clear_user_data_spec.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@
   let!(:conversation) { create(:live_chat_conversation, intervention: intervention) }
   let!(:user_intervention1) { create(:user_intervention, intervention: intervention, user: create(:user, :participant, :confirmed)) }
   let!(:user_intervention2) { create(:user_intervention, intervention: intervention, user: create(:user, :guest, :confirmed)) }
+  let!(:user_intervention3) { create(:user_intervention, intervention: intervention, user: create(:user, :confirmed, roles: ['predefined_participant'])) }
```
Comment (OskarDybowski) [inline spec/jobs/data_clear_jobs/clear_user_data_spec.rb:10]: copied a comment by @WiktoriaRozanska 
> Why did you define that the owner of this intervention is a predefined participant?

File: db/migrate/20230912091140_update_foreign_key_for_user_in_predefined_user_parameter.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class UpdateForeignKeyForUserInPredefinedUserParameter < ActiveRecord::Migration[6.1]
+  def change
+    remove_foreign_key :predefined_user_parameters, :users
+    add_foreign_key :predefined_user_parameters, :users, on_delete: :cascade
+  end
```
Comment (WiktoriaRozanska) [inline db/migrate/20230912091140_update_foreign_key_for_user_in_predefined_user_parameter.rb:5]: It's Okay, but why? 🤔 It is just my curiosity

Comment (WiktoriaRozanska) [reply db/migrate/20230912091140_update_foreign_key_for_user_in_predefined_user_parameter.rb:5]: Give me answer on slack and here 🙏

File: db/schema.rb [hunk 3]
```diff
@@ -177,6 +177,9 @@
     t.string "chart_type", default: "bar_chart"
     t.boolean "trend_line", default: false, null: false
     t.integer "position", default: 1, null: false
+    t.string "interval_type", default: "monthly"
+    t.datetime "date_range_start"
+    t.datetime "date_range_end"
```
Comment (WiktoriaRozanska) [inline db/schema.rb:182]: why is it here?

**Review by WiktoriaRozanska — APPROVED** [review]
🚀

**Pull request merged**